helm/cloud-provider-vsphere/templates/deployment.yaml

```diff
@@ -0,0 +1,96 @@
+{{- if .Values.kamaj.enabled -}}
+{{- $config := .Values.config -}}
+{{- if .Values.global -}}
+{{- if .Values.global.config -}}
+{{- $config = mergeOverwrite (deepCopy .Values.config) .Values.global.config -}}
+{{- end -}}
+{{- end -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "cpi.name" . }}
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+    app: {{ template "cpi.name" . }}
+    vsphere-cpi-infra: deployment
+    component: cloud-controller-manager
+    tier: control-plane
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    {{- if .Values.daemonset.annotations }}
+    {{- toYaml .Values.daemonset.annotations | nindent 4 }}
+    {{- end }}
+spec:
+  replicas: {{ .Values.daemonset.replicaCount }}
```

Comment:
It feels a bit weird to have the `replicas` field under daemonset. Wdyt about moving it under the `kamaji` key for instance?